File: python/leetcode-75/arrays-and-strings/can-place-flowers.py
from typing import List

def canPlaceFlowers(flowerbed: List[int], n: int) -> bool:
        
        for i in range(len(flowerbed)):
            if flowerbed[i] == 0 and (i == 0 or flowerbed[i-1] != 1) and (i == len(flowerbed)-1 or flowerbed[i+1] != 1):
                n -= 1
                flowerbed[i] = 1

        return n <= 0

print(canPlaceFlowers([1,0,0,0,1], 1))

print(canPlaceFlowers([1,0,0,0,1], 2))

File: javascript/leetcode-75/arrays-and-strings/can-place-flowers.js
var canPlaceFlowers = function(flowerbed, n) {

    // we need to check the flowers in a 0 where there are 0s adjacent
    // check in the following way 0 && flowerbed[i - 1] && flowerbed[i + 1]


    // for loop to check values in flowebed array
    for (let i = 0; i < flowerbed.length; i++) {

        // check if a space is free for plant / if current space is 0 and before and after not 1
        if (flowerbed[i] === 0 
            && flowerbed[i - 1] !== 1 
            && flowerbed[i + 1] !== 1) { 
            // subtract from number of unplanted plants
            n--
            // update flowerbed array to show plant is occupying this space
            flowerbed[i] = 1
        }
    }

    // check if plants are all planted return true or false
    return n <= 0
}

console.log(canPlaceFlowers([1,0,0,0,1], 1))

console.log(canPlaceFlowers([1,0,0,0,1], 2))